inference-engine/src/mkldnn_plugin/nodes/mkldnn_normalize_node.cpp

```diff
@@ -812,575 +813,603 @@ void MKLDNNNormalizeL2Node::setPostOps(mkldnn::primitive_attr &attr, bool initWe
             constexpr int align = 16;
-            eltwiseNode->appendPostOps(ops, getOutputShapeAtPort(0).getStaticDims(), align);
+            eltwiseNode->appendPostOps(ops, dims, align);
             continue;
         }
 
         IE_THROW() << "Fusing of " << NameFromType(node->getType()) << " operation to " << NameFromType(this->getType()) << " node is not implemented";
     }
 
-    attr.set_post_ops(ops);
+    kernel_attrs.set_post_ops(ops);
 }
 
 void MKLDNNNormalizeL2Node::createPrimitive() {
     auto& dstMemPtr = getChildEdgeAt(DATA)->getMemoryPtr();
     auto& srcMemPtr = getParentEdgeAt(DATA)->getMemoryPtr();
     if (!dstMemPtr || !dstMemPtr->GetPrimitivePtr())
-        IE_THROW() << errorPrefix << "can't get destination memory";
+        THROW_ERROR << "can't get destination memory";
     if (!srcMemPtr || !srcMemPtr->GetPrimitivePtr())
-        IE_THROW() << errorPrefix << "can't get input memory";
+        THROW_ERROR << "can't get input memory";
     if (getSelectedPrimitiveDescriptor() == nullptr)
-        IE_THROW() << errorPrefix << "has nullable preferable primitive descriptor";
-
-    if (!cornerCase) {
-        auto selectedPD = getSelectedPrimitiveDescriptor();
-        jcp.src_dt = MKLDNNExtensionUtils::IEPrecisionToDataType(selectedPD->getConfig().inConfs[0].desc->getPrecision());
-        jcp.dst_dt = MKLDNNExtensionUtils::IEPrecisionToDataType(selectedPD->getConfig().outConfs[0].desc->getPrecision());
-        jcp.src_data_size = MKLDNNExtensionUtils::sizeOfDataType(jcp.src_dt);
-        jcp.dst_data_size = MKLDNNExtensionUtils::sizeOfDataType(jcp.dst_dt);
-
-        jcp.is_nchw = jcp.is_nhwc = jcp.is_blk = false;
-        if (getParentEdgeAt(0)->getMemory().getDesc().hasLayoutType(LayoutType::ncsp)) {
-            jcp.is_nchw = true;
-        } else if (getParentEdgeAt(0)->getMemory().getDesc().hasLayoutType(LayoutType::nCsp16c) ||
-                  getParentEdgeAt(0)->getMemory().getDesc().hasLayoutType(LayoutType::nCsp8c)) {
-            jcp.is_blk = true;
+        THROW_ERROR << "has nullable preferable primitive descriptor";
+
+    if (!attrs.cornerCase) {
+        if (srcMemPtr->getDesc().hasLayoutType(LayoutType::ncsp)) {
+            attrs.is_nchw = true;
+        } else if (srcMemPtr->getDesc().hasLayoutType(LayoutType::nCsp16c) ||
+                   srcMemPtr->getDesc().hasLayoutType(LayoutType::nCsp8c)) {
+            attrs.is_blk = true;
+        } else if (srcMemPtr->getDesc().hasLayoutType(LayoutType::nspc)) {
+            attrs.is_nhwc = true;
         } else {
-            jcp.is_nhwc = true;
+            THROW_ERROR << "has selected layout which is not supported";
+        }
+    }
+
+    if (inputShapesDefined()) {
+        if (needPrepareParams())
+            prepareParams();
+        updateLastInputDims();
+    }
+}
+
+void MKLDNNNormalizeL2Node::prepareParams() {
+    const auto& dims = getParentEdgeAt(DATA)->getMemoryPtr()->getStaticDims();
```

Comment:
It's incorrect because I get information about layout using `srcMemPtr` in `createPrimitive` before `prepareParams`. So I make necessary checks in `createPrimitive` not in `prepareParams`

Reply:
but you call `prepareParams` on each inference stage and you can get memory error on inference stage